File: validateTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isValidBST(TreeNode* root) {
        return helper(root,NULL,NULL);
    }
    bool helper (TreeNode* node, TreeNode *min, TreeNode *max) {
        
        if(node==NULL) return true;
        
        int value=node->val;
        if (min!=NULL && value<=min->val) return false;
        if (max!=NULL && value>=max->val) return false;
        
        return helper(node->left,min,node) && helper(node->right,node,max);
    }
};

File: Validate Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        
        
        def validBST(root, mini, maxi):
            
            if root is None:
                return True
            if maxi <= root.val or root.val <= mini:
                return False
            
            print(maxi,root.val,mini)
            return validBST(root.left,mini, root.val) and validBST(root.right, root.val, maxi)
            
            
        return validBST(root,float("-inf"),float("inf"))